# COMP5310 Assignment 1 Dataset Audition

## Group 208

This notebook provides a lightweight and comparable assessment of the three
candidate datasets for Assignment 1. The purpose is to compare their size,
variable types, analytical opportunities, data-quality risks, stakeholder
relevance, and ethical or feasibility considerations before selecting one
dataset for detailed preparation and exploratory analysis.

In [2]:
import pandas as pd

In [3]:
train_df = pd.read_csv("../data/raw/train_occupancy.csv")
airline_df = pd.read_csv("../data/raw/airline_delay.csv")
hotel_df = pd.read_csv("../data/raw/hotel_bookings.csv")

In [4]:
print("NSW Train Occupancy:", train_df.shape)
print("Air Travel Delay:", airline_df.shape)
print("Hotel Booking:", hotel_df.shape)

NSW Train Occupancy: (50250, 14)
Air Travel Delay: (101000, 29)
Hotel Booking: (119987, 32)


In [5]:
train_df.head()

,day,Actual.Stop.Station,Actual.Station.Arrv.Time,Actual.Station.Dprt.Time,Segment.Direction,Trip.Name,Service.Line,Orig..Station,Dest..Station,Leading.Set.Type,Node.Seq.Order,Actual.Station.Dprt.Time.Band,Occupancy Status,Occupancy Range
0,9,Miranda,2017-01-10 00:18:17,2017-01-10 00:19:04,Down,620S,Illawarra,Central,Cronulla,T,18,00:15-00:29,MANY_SEATS_AVAILABLE,Low: 0-399
1,13,Beecroft,2017-01-13 08:05:59,2017-01-13 08:06:56,Down,152C,North via Macquarie Park,Central,Hornsby,A,16,08:00-08:14,MANY_SEATS_AVAILABLE,Low: 0-399
2,12,Arncliffe,2017-01-12 07:22:51,2017-01-12 07:23:36,Down,607B,Illawarra,Central,Waterfall,T,6,07:15-07:29,MANY_SEATS_AVAILABLE,Low: 0-399
3,13,Dulwich Hill,2017-01-13 07:20:39,2017-01-13 07:21:53,Down,58-F,Bankstown,Central,Birrong,A,7,07:15-07:29,MANY_SEATS_AVAILABLE,Low: 0-399
4,11,Belmore,2017-01-11 18:38:28,2017-01-11 18:39:07,Up,33-P,Bankstown,Birrong,Central,K,7,18:30-18:44,MANY_SEATS_AVAILABLE,Low: 0-399


In [6]:
airline_df.head()

,Year,Month,DayofMonth,DayOfWeek,DepTime,CRSDepTime,ArrTime,CRSArrTime,UniqueCarrier,FlightNum,...,TaxiIn,TaxiOut,Cancelled,CancellationCode,Diverted,CarrierDelay,WeatherDelay,NASDelay,SecurityDelay,LateAircraftDelay
0,2008,1,31,4,611.0,610,725.0,720,WN,548,...,3.0,13.0,0,NaN,0,NaN,NaN,NaN,NaN,NaN
1,2008,1,18,5,1946.0,1935,2007.0,2010,WN,3770,...,6.0,7.0,0,NaN,0,NaN,NaN,NaN,NaN,NaN
2,2008,1,17,4,1018.0,1020,1102.0,1110,WN,12,...,3.0,9.0,0,NaN,0,NaN,NaN,NaN,NaN,NaN
3,2008,1,10,4,2057.0,2055,2304.0,2300,WN,74,...,5.0,14.0,0,NaN,0,NaN,NaN,NaN,NaN,NaN
4,2008,1,15,2,1825.0,18:25,2031.0,2050,WN,2530,...,5.0,13.0,0,NaN,0,NaN,NaN,NaN,NaN,NaN


In [7]:
hotel_df.head()

,hotel,is_canceled,lead_time,arrival_date_year,arrival_date_month,arrival_date_week_number,arrival_date_day_of_month,stays_in_weekend_nights,stays_in_week_nights,adults,...,deposit_type,agent,company,days_in_waiting_list,customer_type,adr,required_car_parking_spaces,total_of_special_requests,reservation_status,reservation_status_date
0,City Hotel,1,29,2016,February,6,5,6,14,1,...,No Deposit,9.0,NaN,0,Transient,70.91,0,0,Canceled,2016-01-15
1,Resort Hotel,0,312,2017,March,10,5,2,5,2,...,No Deposit,298.0,NaN,0,Transient-Party,56.00,0,0,Check-Out,2017-03-12
2,Resort Hotel,0,19,2016,February,9,27,0,1,2,...,No Deposit,5.0,NaN,0,Transient,0.00,0,0,Check-Out,2016-02-28
3,City Hotel,1,159,2017,July,27,8,1,1,2,...,No Deposit,9.0,NaN,0,Transient,89.10,0,0,Canceled,2017-03-01
4,City Hotel,0,89,2016,June,27,26,2,3,2,...,No Deposit,9.0,NaN,0,Transient,107.10,0,3,Check-Out,2016-07-01


In [8]:
print("TRAIN")
train_df.info()

TRAIN
<class 'pandas.DataFrame'>
RangeIndex: 50250 entries, 0 to 50249
Data columns (total 14 columns):
 #   Column                         Non-Null Count  Dtype
---  ------                         --------------  -----
 0   day                            50250 non-null  int64
 1   Actual.Stop.Station            50176 non-null  str  
 2   Actual.Station.Arrv.Time       50243 non-null  str  
 3   Actual.Station.Dprt.Time       50239 non-null  str  
 4   Segment.Direction              50172 non-null  str  
 5   Trip.Name                      50250 non-null  str  
 6   Service.Line                   50191 non-null  str  
 7   Orig..Station                  50170 non-null  str  
 8   Dest..Station                  50187 non-null  str  
 9   Leading.Set.Type               50173 non-null  str  
 10  Node.Seq.Order                 50250 non-null  int64
 11  Actual.Station.Dprt.Time.Band  50250 non-null  str  
 12  Occupancy Status               50177 non-null  str  
 13  Occupancy Range      

In [9]:
print("AIRLINE")
airline_df.info()

AIRLINE
<class 'pandas.DataFrame'>
RangeIndex: 101000 entries, 0 to 100999
Data columns (total 29 columns):
 #   Column             Non-Null Count   Dtype  
---  ------             --------------   -----  
 0   Year               101000 non-null  int64  
 1   Month              101000 non-null  int64  
 2   DayofMonth         101000 non-null  int64  
 3   DayOfWeek          101000 non-null  int64  
 4   DepTime            99873 non-null   str    
 5   CRSDepTime         101000 non-null  str    
 6   ArrTime            99719 non-null   str    
 7   CRSArrTime         101000 non-null  str    
 8   UniqueCarrier      101000 non-null  str    
 9   FlightNum          101000 non-null  int64  
 10  TailNum            96854 non-null   str    
 11  ActualElapsedTime  99690 non-null   str    
 12  CRSElapsedTime     101000 non-null  str    
 13  AirTime            99690 non-null   str    
 14  ArrDelay           97700 non-null   float64
 15  DepDelay           97850 non-null   float64
 16  Origi

In [10]:
print("HOTEL")
hotel_df.info()

HOTEL
<class 'pandas.DataFrame'>
RangeIndex: 119987 entries, 0 to 119986
Data columns (total 32 columns):
 #   Column                          Non-Null Count   Dtype  
---  ------                          --------------   -----  
 0   hotel                           119987 non-null  str    
 1   is_canceled                     119987 non-null  int64  
 2   lead_time                       119987 non-null  int64  
 3   arrival_date_year               119987 non-null  int64  
 4   arrival_date_month              119987 non-null  str    
 5   arrival_date_week_number        119987 non-null  int64  
 6   arrival_date_day_of_month       119987 non-null  int64  
 7   stays_in_weekend_nights         119987 non-null  int64  
 8   stays_in_week_nights            119987 non-null  int64  
 9   adults                          119987 non-null  int64  
 10  children                        119805 non-null  float64
 11  babies                          119987 non-null  int64  
 12  meal                 

In [11]:
print("TRAIN MISSING VALUES")
print(train_df.isna().sum())

TRAIN MISSING VALUES
day                               0
Actual.Stop.Station              74
Actual.Station.Arrv.Time          7
Actual.Station.Dprt.Time         11
Segment.Direction                78
Trip.Name                         0
Service.Line                     59
Orig..Station                    80
Dest..Station                    63
Leading.Set.Type                 77
Node.Seq.Order                    0
Actual.Station.Dprt.Time.Band     0
Occupancy Status                 73
Occupancy Range                   0
dtype: int64


In [12]:
print("AIRLINE MISSING VALUES")
print(airline_df.isna().sum())

AIRLINE MISSING VALUES
Year                     0
Month                    0
DayofMonth               0
DayOfWeek                0
DepTime               1127
CRSDepTime               0
ArrTime               1281
CRSArrTime               0
UniqueCarrier            0
FlightNum                0
TailNum               4146
ActualElapsedTime     1310
CRSElapsedTime           0
AirTime               1310
ArrDelay              3300
DepDelay              3150
Origin                   0
Dest                     0
Distance                 0
TaxiIn                2311
TaxiOut               2153
Cancelled                0
CancellationCode     99364
Diverted                 0
CarrierDelay         81177
WeatherDelay         81177
NASDelay             81177
SecurityDelay        81177
LateAircraftDelay    81177
dtype: int64


In [13]:
print("HOTEL MISSING VALUES")
print(hotel_df.isna().sum())

HOTEL MISSING VALUES
hotel                                  0
is_canceled                            0
lead_time                              0
arrival_date_year                      0
arrival_date_month                     0
arrival_date_week_number               0
arrival_date_day_of_month              0
stays_in_weekend_nights                0
stays_in_week_nights                   0
adults                                 0
children                             182
babies                                 0
meal                                 138
country                              628
market_segment                       129
distribution_channel                 164
is_repeated_guest                      0
previous_cancellations                 0
previous_bookings_not_canceled         0
reserved_room_type                     0
assigned_room_type                     0
booking_changes                        0
deposit_type                           0
agent                              1

In [14]:
print("Train duplicates:", train_df.duplicated().sum())
print("Airline duplicates:", airline_df.duplicated().sum())
print("Hotel duplicates:", hotel_df.duplicated().sum())

Train duplicates: 250
Airline duplicates: 1000
Hotel duplicates: 31328


In [15]:
train_df.nunique().sort_values()

Occupancy Range                      3
Segment.Direction                    4
Occupancy Status                     6
day                                  8
Leading.Set.Type                    10
Service.Line                        30
Node.Seq.Order                      38
Orig..Station                       67
Dest..Station                       68
Actual.Station.Dprt.Time.Band       96
Actual.Stop.Station                307
Trip.Name                         3587
Actual.Station.Dprt.Time         43139
Actual.Station.Arrv.Time         47198
dtype: int64

In [16]:
airline_df.nunique().sort_values()

Year                    1
Month                   1
CancellationCode        3
Diverted                6
Cancelled               6
DayOfWeek               7
UniqueCarrier          10
DayofMonth             31
SecurityDelay          38
TaxiIn                 56
TaxiOut               113
WeatherDelay          169
NASDelay              191
CarrierDelay          209
CRSElapsedTime        273
LateAircraftDelay     287
Origin                308
Dest                  312
DepDelay              351
ArrDelay              389
AirTime               579
ActualElapsedTime     581
TailNum               622
CRSDepTime            711
Distance              769
CRSArrTime            952
DepTime              2152
ArrTime              2271
FlightNum            2669
dtype: int64

In [17]:
hotel_df.nunique().sort_values()

is_canceled                          2
is_repeated_guest                    2
arrival_date_year                    3
deposit_type                         3
hotel                                4
required_car_parking_spaces          5
babies                               5
children                             5
total_of_special_requests            6
reservation_status                   6
customer_type                        8
distribution_channel                 9
meal                                10
reserved_room_type                  10
arrival_date_month                  12
assigned_room_type                  12
market_segment                      14
adults                              14
previous_cancellations              15
stays_in_weekend_nights             17
booking_changes                     21
arrival_date_day_of_month           31
stays_in_week_nights                35
arrival_date_week_number            53
previous_bookings_not_canceled      73
days_in_waiting_list     

In [18]:
missing_summary = pd.DataFrame({
    "Train": train_df.isna().sum(),
    "Airline": airline_df.isna().sum(),
    "Hotel": hotel_df.isna().sum()
})

missing_summary["Train_%"] = (
    missing_summary["Train"] / len(train_df) * 100
)

missing_summary["Airline_%"] = (
    missing_summary["Airline"] / len(airline_df) * 100
)

missing_summary["Hotel_%"] = (
    missing_summary["Hotel"] / len(hotel_df) * 100
)

missing_summary

,Train,Airline,Hotel,Train_%,Airline_%,Hotel_%
Actual.Station.Arrv.Time,7.0,NaN,NaN,0.013930,NaN,NaN
Actual.Station.Dprt.Time,11.0,NaN,NaN,0.021891,NaN,NaN
Actual.Station.Dprt.Time.Band,0.0,NaN,NaN,0.000000,NaN,NaN
Actual.Stop.Station,74.0,NaN,NaN,0.147264,NaN,NaN
ActualElapsedTime,NaN,1310.0,NaN,NaN,1.29703,NaN
...,...,...,...,...,...,...
reservation_status_date,NaN,NaN,0.0,NaN,NaN,0.0
reserved_room_type,NaN,NaN,0.0,NaN,NaN,0.0
stays_in_week_nights,NaN,NaN,0.0,NaN,NaN,0.0
stays_in_weekend_nights,NaN,NaN,0.0,NaN,NaN,0.0


In [19]:
print("TRAIN - highest missingness")
print(
    (train_df.isna().mean() * 100)
    .sort_values(ascending=False)
    .head(10)
)

print("\nAIRLINE - highest missingness")
print(
    (airline_df.isna().mean() * 100)
    .sort_values(ascending=False)
    .head(10)
)

print("\nHOTEL - highest missingness")
print(
    (hotel_df.isna().mean() * 100)
    .sort_values(ascending=False)
    .head(10)
)

TRAIN - highest missingness
Orig..Station               0.159204
Segment.Direction           0.155224
Leading.Set.Type            0.153234
Actual.Stop.Station         0.147264
Occupancy Status            0.145274
Dest..Station               0.125373
Service.Line                0.117413
Actual.Station.Dprt.Time    0.021891
Actual.Station.Arrv.Time    0.013930
day                         0.000000
dtype: float64

AIRLINE - highest missingness
CancellationCode     98.380198
LateAircraftDelay    80.373267
SecurityDelay        80.373267
NASDelay             80.373267
WeatherDelay         80.373267
CarrierDelay         80.373267
TailNum               4.104950
ArrDelay              3.267327
DepDelay              3.118812
TaxiIn                2.288119
dtype: float64

HOTEL - highest missingness
company                 94.311884
agent                   13.785660
country                  0.523390
children                 0.151683
distribution_channel     0.136681
customer_type            0.12834

In [20]:
print("Train duplicate %:", train_df.duplicated().mean() * 100)
print("Airline duplicate %:", airline_df.duplicated().mean() * 100)
print("Hotel duplicate %:", hotel_df.duplicated().mean() * 100)

Train duplicate %: 0.4975124378109453
Airline duplicate %: 0.9900990099009901
Hotel duplicate %: 26.109495195312824


In [21]:
comparison = pd.DataFrame({
    "Dataset": [
        "NSW Train Occupancy",
        "Air Travel Delay",
        "Hotel Booking"
    ],
    
    "Rows": [
        len(train_df),
        len(airline_df),
        len(hotel_df)
    ],
    
    "Columns": [
        len(train_df.columns),
        len(airline_df.columns),
        len(hotel_df.columns)
    ],
    
    "Variable Types": [
        "2 numeric, 12 categorical/text",
        "14 numeric, 15 categorical/text",
        "20 numeric, 12 categorical/text"
    ],
    
    "Duplicate %": [
        train_df.duplicated().mean() * 100,
        airline_df.duplicated().mean() * 100,
        hotel_df.duplicated().mean() * 100
    ]
})

comparison

,Dataset,Rows,Columns,Variable Types,Duplicate %
0,NSW Train Occupancy,50250,14,"2 numeric, 12 categorical/text",0.497512
1,Air Travel Delay,101000,29,"14 numeric, 15 categorical/text",0.990099
2,Hotel Booking,119987,32,"20 numeric, 12 categorical/text",26.109495


## Dataset comparison and selection

The three candidate datasets differ in size, analytical opportunities, and data-quality risks. The NSW Train Occupancy dataset contains 50,250 observations and 14 variables, with relatively low missingness and approximately 0.5% exact duplicate rows. The Air Travel Delay dataset contains 101,000 observations and 29 variables, providing substantial opportunities to investigate delay patterns, but several delay-related variables have high levels of missingness. The Hotel Booking dataset contains 119,987 observations and 32 variables and offers rich information about booking behaviour, but approximately 26.1% of its rows are exact duplicates, creating a substantial data-quality concern.

The NSW Train Occupancy dataset is selected for further analysis because it provides a relatively clean and manageable evidence base while directly supporting the assignment mission of advising a transport operations team about crowding patterns and opportunities for improved service planning. Its variables describe train service characteristics, station information, timing, occupancy status and occupancy range, providing opportunities to investigate when and where crowding occurs.